### ISSUE TITLE
Incorrect TooManyTemplateParams

### ISSUE THREAD

**Issue by tux-rampage** [vimeo/psalm#3111]
When using the AMP libraries, which do have Psalm-Annotations, Psalm will complain about too many template params.

### Versions:

 - amphp/amp: 2.4.2
 - amphp/parallel: 1.3.0
 - psalm: 3.10.1

### Example:

Amp's Promise with template annotation: https://github.com/amphp/amp/blob/master/lib/Promise.php
Amp Parallel's worker interface with template parameterized Promise: https://github.com/amphp/parallel/blob/7f00b8effb4fb439176aba0fa392abd120cbd952/lib/Worker/Worker.php#L37

```php
/** @param \Amp\Parallel\Worker\Worker $worker */
function test(\Amp\Parallel\Worker\Worker $worker): void
{
    $worker->shutdown(); // Psalm will raise a TooManyTemplateParams error for this call
}
```

### Expected Behavior

No errors

### Actual behavior

ERROR: TooManyTemplateParams (...) - Amp\Promise<int> has too many template params, expecting 0 (see https://psalm.dev/184)

**Comment by tux-rampage** [comment]
This is not reproducable in psalm.dev: https://psalm.dev/r/7147d5b053
It must be a problem when the consumed types are located in a composer dependency.

**Comment by muglug** [comment]
This should hopefully be fixed in latest Psalm.

**Comment by tux-rampage** [comment]
Thanks. I'll give it a try.

**Comment by tux-rampage** [comment]
:+1: Can confirm this is fixed in psalm 3.11.2
